src/core/unit.c

```diff
@@ -5559,6 +5563,9 @@ void unit_export_state_files(Unit *u) {
 
         (void) unit_export_invocation_id(u);
 
+        if (!MANAGER_IS_SYSTEM(u->manager))
+                return;
+
         c = unit_get_exec_context(u);
         if (c) {
```

Comment:
For feature parity sake it might make sense to also allow exporting these fields for user units. But since they seem more "dangerous" than just adding invocation IDs to journal logs I opted not to do anything about it right now.